regression-test/suites/nereids_p0/sql_functions/string_functions/test_string_function.groovy

```diff
@@ -188,6 +188,31 @@ suite("test_string_function") {
     qt_sql "select elt(1, \"hello\", \"doris\");"
     qt_sql "select elt(2, \"hello\", \"doris\");"
     qt_sql "select elt(3, \"hello\", \"doris\");"
+    qt_sql """
+    SELECT t1.no
+      ,t1.sub_str
+      ,t1.str
+      ,substring_index(t1.str, t1.sub_str, -1)
```

Comment:
should write 2 cases:
1. insert the data into a table and then use select statement to test it.
2. test constant value